concordia/forms.py

```diff
@@ -23,6 +53,22 @@ class Meta(RegistrationForm.Meta):
         }
 
 
+class UserLoginForm(AuthenticationForm):
+    def confirm_login_allowed(self, user):
+        # If the user provided a correct username and password combination,
+        # but has not yet confirmed their email,
+        # resend the email activation request and display a custom message.
+        if not user.is_active:
+            logger.warning("Inactive user tried to log in with valid credentials.")
+            view = RegistrationView(request=self.request)
+            view.send_activation_email(user)
+
+            raise forms.ValidationError(
+                "Please check your email and click the link to activate your account.",
```

Comment:
@elainekamlley @lalgee Would like your suggestions for wording here. This is the message that is displayed when someone provides valid credentials but the account is inactive because they haven't confirmed their email. Just before this message is displayed, the activation email is resent. Maybe we should also say something like if you don't get the email, please contact us. Maybe we should also mask the email address and say "email was sent to r***y@whatever.com" or something like that as well.

Reply:
I like the idea of showing a masked email address to help ensure they're using the right account info. And also stating explicitly that they did not yet activate.  Suggestion:

"This account has not yet been activated.  An activation email has been sent to r***y@aol.com. Please check for this message and click the link to finish your account registration."

Reply:
I like showing the address but wouldn't bother masking it since it's a value which they just entered so we're not making otherwise-hidden information visible and that might give them a chance to notice a typo.

Reply:
I agree with Chris, here. Good spot check opportunity for the user.

Reply:
This is for the login page, not the forgot password page. The problem is that if they try to sign in with username and password, rather than email and password, we could be leaking email addresses if we display them.